Nadim.CinemaReservationSystem.Web/Services/FilmService.cs

```diff
@@ -34,6 +35,26 @@ private bool FilmInfoValid(FilmInfo filmInfo)
                 && filmInfo.Duration != 0;
         }
 
+        private IQueryable<Session> CreateFilteredFilmSessionsQuery(int filmId, FilmFilter filter)
+        {
+            IQueryable<Session> query = dbContext.Sessions
```

Comment:
In EF for data that isn't going to change during context lifetime it's preferable to use AsNoTracking.